### PR TITLE
Version file management

### LINKED ISSUE #326: Any downloaded version - latest version (if equal filenames was used)..
Browser & Version: `Firefox 57.0.3-1`

Operating System: `Arch Linux`

(I dont think info above will be helpful, but it was in the template... xD)

Steps to reproduce:

1. Download any version of plugin - it will be latest version...
Video: https://youtu.be/RAFGkm3BIRo

### PR DESCRIPTION
**Pull request by phase**
File name conflicts can happen if they're all in the same version.

I tested with creating a project, downloading the version, renaming the
project, downloading the version again, and deleting the project.
Everything was good.

Fixes #326
Closes #327

I'm working on a script to fix the prod file paths. Don't merge until it's in.

### CODE REVIEW

File: app/db/impl/access/ProjectBase.scala [hunk 1]
```diff
@@ -149,11 +150,12 @@ class ProjectBase(override val service: ModelService,
 
     channel.remove()
 
+    val p = this.fileManager.getProjectDir(proj.ownerName, proj.name).resolve(channel.name)
```
Comment (kashike) [inline app/db/impl/access/ProjectBase.scala]: nitpick: please don't use a single character for a variable name

Comment (phase) [reply app/db/impl/access/ProjectBase.scala]: that's a temp, I wanted to see if that was being called

**Comment by phase** [comment]
Closes #328 too.

:tada: :tada: :tada: :tada: 

<sup><sub>Anything else you guys want to break while I'm here?</sub></sup>

**Comment by phase** [comment]
I fixed up some issues @felixoi notified me of.

File: scripts/fix_file_paths_b725a7c.py3 [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      os.rename(old_file_path, new_file_path)
+      if os.path.isfile(old_sig_path):
+        # Move Sig
+        println("Moving signature " + old_sig_path + " to " + new_sig_path)
+        os.rename(old_sig_path, new_sig_path)
+    else:
+      # This probably means someone uploaded a version that conflicted with another
+      # version and it got overwritten. I shouldn't be deleted because it might match
+      # another version in the DB
+      println("ERROR: " + old_file_path + " doesn't have the hash " + hash)
```
Comment (progwml6) [inline scripts/fix_file_paths_b725a7c.py3]: we are checking the MD5's if this doesn't match the current version it is one of the versions affected by the bug and needs to be removed from the DB

File: scripts/fix_file_paths_b725a7c.py3 [hunk 3]
```diff
@@ -0,0 +1,83 @@
+# File path fixing for Ore
+# Run fix_file_paths_ecd8100.py3 first
+#
+# b725a7cce20920dbc07530ac5b60569f3ba54ebe changed the
+# version directories. This script will move the files
+# into separate directories.
+#
+# by @phase
```
Comment (progwml6) [inline scripts/fix_file_paths_b725a7c.py3]: felixoi and I worked on this one also

**Review by progwml6 — CHANGES_REQUESTED** [review]
the newer file should have a different name like what the early version I sent you had was called, keep the hashes in them that is probably useful

File: scripts/fix_file_paths_b725a7c.py3 (hunk 2 above)
Comment (phase) [reply scripts/fix_file_paths_b725a7c.py3]: Done in the latest commit

**Comment by phase** [comment]
This good to go?

**Comment by progwml6** [comment]
The readme needs to be fixed for the rename

**Comment by phase** [comment]
Fixed

**Pull request merged**